web/ui/dashboard/src/app/private/pages/project/endpoints/endpoints.component.ts

```diff
@@ -56,29 +55,40 @@ import { LoaderModule } from 'src/app/private/components/loader/loader.module';
 		PermissionDirective,
 		EndpointSecretComponent,
 		DeleteModalComponent,
-		LoaderModule
+		LoaderModule,
+		DialogDirective
 	],
 	templateUrl: './endpoints.component.html',
 	styleUrls: ['./endpoints.component.scss']
 })
 export class EndpointsComponent implements OnInit {
+	@ViewChild('endpointDialog', { static: true }) endpointDialog!: ElementRef<HTMLDialogElement>;
+	@ViewChild('secretDialog', { static: true }) secretDialog!: ElementRef<HTMLDialogElement>;
+	@ViewChild('deleteDialog', { static: true }) deleteDialog!: ElementRef<HTMLDialogElement>;
+
 	showCreateEndpointModal = this.router.url.split('/')[4] === 'new';
 	showEditEndpointModal = this.router.url.split('/')[5] === 'edit';
 	endpointsTableHead = ['Name', 'Status', 'ID', '', '', ''];
 	displayedEndpoints?: { date: string; content: ENDPOINT[] }[];
 	endpoints?: { pagination?: PAGINATION; content?: ENDPOINT[] };
 	selectedEndpoint?: ENDPOINT;
 	isLoadingEndpoints = true;
-	showEndpointSecret = false;
 	isDeletingEndpoint = false;
 	showDeleteModal = false;
 	isTogglingEndpoint = false;
 	isSendingTestEvent = false;
 	endpointSearchString!: string;
+	action: 'create' | 'update' = 'create';
 
 	constructor(public router: Router, public privateService: PrivateService, public projectService: ProjectService, private endpointService: EndpointsService, private generalService: GeneralService, public route: ActivatedRoute) {}
 
 	ngOnInit() {
+		const urlParam = this.route.snapshot.params.id;
+		if (urlParam) {
+			urlParam === 'new' ? (this.action = 'create') : (this.action = 'update');
+			this.endpointDialog.nativeElement.showModal();
+		}
+
```

Comment:
This implementation here, why?

We already have a route [`endpoints/:id`](https://github.com/frain-dev/convoy/blob/e991ece4a33c4a8b1977359ff9b5a9581649cdd2/web/ui/dashboard/src/app/private/pages/project/project.module.ts#L72) that handles this?